extensions/cl_khr_unified_svm.asciidoc

```diff
@@ -0,0 +1,1081 @@
++
+--
+`RESOLVED`: Yes, we should.
+--
+
+. Should we add explicit mem alloc flags for `CACHED` and `UNCACHED`?
++
+--
+*UNRESOLVED*:
+Could be specific capabilities rather than mem alloc flags.
+Solve (or at least have explored a layered extension) for the final spec.
+--
+
+. At least for HOST and SHARED allocations, should we have separate mem alloc flags for the host and the device?
++
+--
+`RESOLVED`: We removed the _flags_ argument entirely.
+--
+
+. What are invalid values for `ptr` and `size` for *clEnqueueSVMMigrateMem*?
+How about *clEnqueueSVMMemFill* and *clEnqueueSVMMemcpy*?
+Specifically, is `NULL` a valid value for `ptr`?
+Is `size` equal to zero valid?
++
+--
+*UNRESOLVED*:
+--
+
+. Should we add a device query for a maximum supported SVM alignment, or should the maximum supported alignment implicitly be defined by the size of the largest data type supported by the device?
+Should we allow implementation-defined behavior for alignments larger than the size of the largest data type supported by the device?
++
+--
+*UNRESOLVED*:
+A device query would allow for larger supported alignments, such as page alignment.
+Note that supported alignments should always be a power of two.
+
+Note that there are no maximum supported alignments defined for `posix_memalign` or `_aligned_alloc`, and supported alignments for the standard `aligned_alloc` and `std::aligned_alloc` are implementation-defined.
+
+Suggest adding a device query and use it to determine the maximum supported alignment error code.
+--
+
+. Should we add a device query for a maximum supported SVM fill pattern size, or should the maximum supported fill pattern size implicitly be defined by the size of the largest data type supported by the device?
++
+--
+`RESOLVED`:
+The initial version of this extension will not support larger fill patterns.
+--
+
+. Can a pointer to a device, host, or shared SVM allocation be used to create a `cl_mem` using `CL_MEM_USE_HOST_PTR`?
++
+--
+*UNRESOLVED*:
+Trending "no" in all cases.
+If the SVM allocation is from the same context this could be an error, such as `CL_INVALID_HOST_PTR`.
+If the SVM allocation is from a different context then behavior could be undefined.
+--
+
+. Can a pointer to a device, host, or shared SVM allocation be used to create a `cl_mem` buffer using `CL_MEM_COPY_HOST_PTR`?
++
+--
+*UNRESOLVED*:
+Trending "no" for device and shared USM allocations.
+If the USM allocation is from the same context this could be an error, such as `CL_INVALID_HOST_PTR`.
+If the USM allocation is from a different context then behavior could be undefined.
+
+Trending "yes" for host USM allocations, both when the host USM allocation is from this context and from another context.
+--
+
+. Can a pointer to a device, host, or shared SVM allocation be passed to API functions to read from or write to `cl_mem` objects, such as *clEnqueueReadBuffer* or *clEnqueueWriteImage*?
```

Comment:
Discussed briefly in the January 7th memory subgroup.  We may want to consider buffers and images separately.